class.json-api-endpoints.php

```diff
@@ -1324,8 +1389,20 @@ function load_theme_functions() {
 
 		// since the stuff we care about (CPTS, post formats, are usually on setup or init hooks, we want to load those)
 		$this->copy_hooks( 'after_setup_theme', 'restapi_theme_after_setup_theme', WP_CONTENT_DIR . '/themes' );
+		/**
+		 * Fires after the theme is initialized.
+		 * @see https://codex.wordpress.org/Plugin_API/Action_Reference/after_setup_theme
+		 *
+		 * @since 3.2.0
+		 */
 		do_action( 'restapi_theme_after_setup_theme' );
 		$this->copy_hooks( 'init', 'restapi_theme_init', WP_CONTENT_DIR . '/themes' );
+		/**
+		 * Fires on init, after WordPress has finished loading but before any headers are sent.
```

Comment:
This doesn't fire on `init` per se, but mimics the init hook for theme code being accessed via the REST API (if a theme is declaring a CPT or something like that), if I understand this correctly.